_posts/2017-01-31-git_v_PhpStormu.md

```diff
@@ -0,0 +1,57 @@
+
+#Praxe - jdeme na to!
+Jestli se chceš s gitem naučit v příkazovém řádku (což je těžší, ale užitečnější), přeskoč na [další článek, který vyjde týden po tomto](http://jakpsatphp.cz/git-v-prikazovem-radku). Zbytek článku je psán pro PhpStorm 2016, práce v novější verzi ale bude velmi podobná. 
+
+> Všechny praktické rady budou formátovány tímto způsobem - pokud chceš rychle načíst jen teorii, tyto pasáže přeskakuj - ovšem nedoporučuju to, protože půlka mojí práce potom přijde nazmar. 
+
+##Repozitář
+Místo, kde se všechna gitovská magie odehrává, se nazývá **repozitář** nebo i **repo**. Repozitář je složka v počítači, kterou git sleduje. Dá se určit, že se nějaké soubory sledovat nemají (například lokální konfigurace), nemůžeme ale do repozitáře zahrnout soubory mimo tuto složku. 
+
+> Založ si nový gitový repozitář, tedy postupně: vytvoř si nový projekt v PhpStorm a vyber z horní nabídky _VCS_ (zkratka z version control systems) > _Enable Version Controll Integration..._, tam vyber _git_ a potvrď. Tím se vytvořil gitový repozitář pro celý projekt. Když se teď podíváš do složky s projektem, objevila se tam nová složka _.git_ Pozor, je označená systémem jako skrytá, takže bude viditelná až po povolení zobrazování skrytých souborů. Jestli ji nevidíš, nemusíš to moc řešit, není to příliš důležité. Když tuto složku smažeš, efektivně tím zrušíš sledování repozitáře, kde složka _.git_ byla. 
+
+##Commit
+Základním kamenem pro práci s gitem je **commit**. Commit je zachycení stavu souborů tak, jak zrovna jsou. Commit jako takový není dál dělitelný, je tou nejmenší jednotkou, s kterou se v gitu potkáš. Commity se řadí za sebou v čase, jeden následuje druhý tak, jak byly postupně pořizovány. 
+
+> Vytvoř si nový soubor `index.php`. V tomto momentě by měl PhpStorm nabídnout přidání souboru do verzovacího systému. Pokud ne, klikneš pravým myšítkem na soubor a z místní nabídky vyber _Git_ > _Add_. Název přidaného souboru se defaultně zbarví z červené do zelené - to značí, že je soubor nově přidaný do verzovacího systému. 
+> Do `index.php` zapiš obligátní pozdrav celému světu, aby si měli dál s čím pracovat. Vložíme tedy třeba
+ 
+ ```php
+ <?php echo('Hello World!');
+ ```
+ 
+> A teď soubor ulož v čase - slangově commitni. Zvol z horního menu _VCS_ > _Commit changes_ (zkratka _Ctrl + K_). Otevře se okno, do kterého v dolní části vlož popis commitu (commit message), třeba `první commit`. Okno potvrď a je to! 
+
+Správné vymyšlení názvu a popisu commitu (commit message) je celkem náročné. Jsou to totiž zásadní informace, podle kterých se potom v commitech dá nebo nedá vyznat. Název commitu by měl být výstižný a vyjadřovat, co commit s projektem dělá. Dobré popisy jsou třeba: `added sending request via mail` nebo: `fixed bug #34`. Naopak se vyhni popisům typu: `opraveno` nebo: `další commit`. Název commitu by se měl vejít do 50 znaků, do popisu commitu potom patří zbytek. Podrobnější věci ale nepatří ani tam, protože ty si čtenář může najít přímo v kódu. 
```

Comment:
```fixed bug #34``` touche - jaký jiný příklad bys použil?
```Podrobnější věci...``` chci se opřít do toho, aby tam zbytečně dalece nerozepisovali, co všechno se tam upravovalo a jaké to teď je. Formuloval bys to tedy jinak?